### PR TITLE
TWAP Query class

### PR DESCRIPTION
**Pull request by tallywiesenberg**

### CODE REVIEW

File: src/telliot_feeds/queries/price/twap.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    for s in SPOT_PRICE_PAIRS:
+        asset_id, currency = s.split("/")
+        pairs.append((asset_id.lower(), currency.lower()))
+
+    return pairs
+
+
+@dataclass
+class TWAP(AbiQuery):
+    """Returns the spot price of a cryptocurrency asset in the given currency.
```
Comment (oraclown) [inline src/telliot_feeds/queries/price/twap.py:32]: Add link to the corresponding query type in `/dataSpecs`

File: src/telliot_feeds/queries/price/twap.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+from telliot_feeds.dtypes.value_type import ValueType
+from telliot_feeds.queries.abi_query import AbiQuery
+from telliot_feeds.queries.price.spot_price import CURRENCIES
+from telliot_feeds.queries.price.spot_price import SPOT_PRICE_PAIRS
+
+
+logger = logging.getLogger(__name__)
+
+
+def format_spot_price_pairs() -> List[Tuple[str, str]]:
```
Comment (oraclown) [inline src/telliot_feeds/queries/price/twap.py:19]: Is this a repeat of a function created elsewhere that instead can just be imported?

**Review by oraclown — CHANGES_REQUESTED** [review]
Nice stuff, @tallywiesenberg !

- Add in tests for encoding / decoding the query type's response type
- Create and reference an issue for supporting this new query type (There's a default issue template for this that you'll see)

File: src/telliot_feeds/queries/price/twap.py (hunk 1 above)
Comment (tallywiesenberg) [reply src/telliot_feeds/queries/price/twap.py:32]: Noted

File: src/telliot_feeds/queries/price/twap.py (hunk 2 above)
Comment (tallywiesenberg) [reply src/telliot_feeds/queries/price/twap.py:19]: Indeed!

File: tests/queries/test_query_twap.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    q = TWAP(asset="btc", currency="USD", timespan=86400)
+
+    exp_data_abi = "000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000004545741500000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000151800000000000000000000000000000000000000000000000000000000000000003627463000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000037573640000000000000000000000000000000000000000000000000000000000"  # noqa: E501
+    assert q.query_data.hex() == exp_data_abi
+
+    exp = "18043d2ec387f70a62430753cb717e4f0c8a4de5344c75b3cd8bc65147c7c4c6"
+    assert q.query_id.hex() == exp
+
+
+def test_invalid_currency():
```
Comment (oraclown) [inline tests/queries/test_query_twap.py:22]: Suggestion, not required: you could put `test_invalid_currency` & `test_invalid_pair` & call it test_invalid_params

File: tests/queries/test_query_twap.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def test_invalid_pair():
+
+    with pytest.raises(ValueError):
+        _ = TWAP(asset="xxx", currency="usd", timespan=86400)
+
+
+def test_vsq_usd_twap_price():
+    q = TWAP(asset="vsq", currency="usd", timespan=86400)
```
Comment (oraclown) [inline tests/queries/test_query_twap.py:34]: You can delete `test_vsq_usd_twap_price` & `test_bct_usd_twap_price` bc they're just testing the constructor/query_id, which is already done above.

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks @tallywiesenberg! , just left a couple change requests & suggestions

**Comment by oraclown** [comment]
Related: #265

**Review by oraclown — APPROVED** [review]

**Pull request merged**